sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -544,6 +544,14 @@ abstract class JdbcDialect extends Serializable with Logging {
     if (limit > 0 ) s"LIMIT $limit" else ""
   }
 
+  /**
+   * MS SQL Server version of `getLimitClause`.
+   * This is only supported by SQL Server as it uses TOP (N) instead.
+   */
+  def getTopExpression(limit: Integer): String = {
```

Comment:
This API looks a little hack.

https://github.com/apache/spark/pull/39667 refactor the API and this PR could override getSQLText.
cc @cloud-fan @huaxingao

Reply:
What exactly is hacky in this API? As far as I can see, it is just follows the existing JDBC code. If you are referring to an alternative implementation, then I am happy to consider it, otherwise please be elaborate.

Reply:
I also don't see how the PR you linked solves this problem. Yes, there is no top reference in the general JdbcDialects but now MSSQL dialect needs to reimplement `getSQLText` method which is suboptimal IMHO.

Reply:
I think we should define the different syntax by dialect themself. I exact the API getSQLText so as some dialect could implement it in special way.

Reply:
I agree, it would be good to have something more flexible than the current API; however, I also don't think that moving `getSQLText` into `JdbcDialects` is good approach either: difficult to evolve API, too much boilerplate to copy-paste if you need to change a small detail in the query.

I think it would be good to split the query into fragments and have methods for each. However, there has already been a precedent with `getTableSample` which is similar to TOP N but is only supported by Postgres.

Reply:
I can work on a refined API for JdbcDialects as a follow-up for this work, would that work? I may need to prepare a small SPIP/design doc on it. 

@beliefer @srowen @dongjoon-hyun What do you think? Or should I just work on a new API right away without this change? I would appreciate your suggestions.

Reply:
> I agree, it would be good to have something more flexible than the current API; however, I also don't think that moving `getSQLText` into `JdbcDialects` is good approach either: difficult to evolve API, too much boilerplate to copy-paste if you need to change a small detail in the query.

@sadikovi You said right. I simplify the https://github.com/apache/spark/pull/39667. `getSQLText` only used to organize the SQL syntax.